### ISSUE TITLE
add content related folders as git submodules

### ISSUE THREAD

**Issue by devopstoday11** [intelops/external-docs#72]
@tfsumon ,
After finishing the versioning change task, I need this to be done. Or, if it is quick steps to add in config.toml, let me know what to do, I will do it myself. 

Assuming you finished the versioning task...
This is what I am planning to do to handle the Topics' related folders from outside the docs repo. 

I want to create separate git repos for documentation purposes for the Topics "compage , capten , kubviz , genval". In those independent repos, we will write & submit all the documentation. 
Now, in our main docs repo (external-docs) , I want to add those independent repos as git submodules. 

separate repos -
```
https://github.com/intelops/compage-docs
https://github.com/intelops/capten-docs
https://github.com/intelops/kubviz-docs
https://github.com/intelops/genval-docs
https://github.com/intelops/compage-changelog
https://github.com/intelops/capten-changelog
https://github.com/intelops/kubviz-changelog
https://github.com/intelops/genval-changelog
```

After adding them as git submodules to the main docs repo "external-docs" , the folder names should still be same as original names in this main repo. No need to change names.  This means we don't want to change anything in this main docs repo. 
```
content/compage
content/capten
content/kubviz
content/genval
content/changelog/capten
content/changelog/compage
content/changelog/kubviz
content/changelog/genval
```

Please let me know if you understand the requirements or have any questions. @tfsumon

**Comment by tfsumon** [comment]
This task may require some time and I will handle it.

To begin, please create all topics and changelog repositories based on the current file structure, as illustrated below. Once done, kindly notify me:

changelog >
      compage >
        version_number

compage >
  version_number

**Comment by devopstoday11** [comment]
Ok. We can work on this later. I will try one idea I have first and then I'll let you know when to start working on this one. @tfsumon 
Thanks for prompt response

**Comment by devopstoday11** [comment]
> This task may require some time and I will handle it.
> 
> To begin, please create all topics and changelog repositories based on the current file structure, as illustrated below. Once done, kindly notify me:
> 
> changelog > compage > version_number
> 
> compage > version_number

@tfsumon , I created separate repos like this

https://github.com/intelops/compage-docs/
https://github.com/intelops/compage-changelog/

I created like that for all the topics. 

https://github.com/orgs/intelops/repositories?q=docs&type=all&language=&sort=

**Comment by tfsumon** [comment]
>  All those repos are empty for now.

Kindly add data on those repo so that I can incorporate them as submodules. Submodules won't serve their purpose without any data.

**Comment by devopstoday11** [comment]
> > All those repos are empty for now.
> 
> Kindly add data on those repo so that I can incorporate them as submodules. Submodules won't serve their purpose without any data.
> 
> @devopstoday11

What should be the folders in those repos? @tfsumon 
Give me some examples please.

**Comment by tfsumon** [comment]
Take a look at the following repository. I've added 'compage' 'capten' changelog & docs  as submodules. Let me know if you'd like something similar or if you have other preferences.

https://github.com/tfsumon/sub-external-docs

@devopstoday11

**Comment by devopstoday11** [comment]
> Take a look at the following repository. I've added 'compage' 'capten' changelog & docs as submodules. Let me know if you'd like something similar or if you have other preferences.
> 
> https://github.com/tfsumon/sub-external-docs
> 
> @devopstoday11
> 
> Clone the repository with the following command to include all its submodule files and folders:
> 
> `git clone --recursive https://github.com/tfsumon/sub-external-docs.git`

Yeah I got the idea now.

One more idea I would like to discuss with you :

can I keep both docs and changelog of a topic in same repo (new separate repo) and still be able to add it as submodules to the correct paths we want into the main docs site repo? @tfsumon

**Comment by tfsumon** [comment]
It's not possible to achieve that. Currently, all changelogs in the main repository are situated inside the 'changelog' folder, and topics are within the 'content' folder. Git submodules do not provide an option to include a specific repository folder in a particular location.

But it is possible if you organize all topic changelogs inside the 'content' folder instead of the 'changelog' folder.

@devopstoday11

**Comment by devopstoday11** [comment]
> But it is possible if you organize all topic changelogs inside the 'content' folder instead of the 'changelog' folder.

could you please show me what you meant exactly? I didn't get it.
give me example folder structure in this message how to organize the folders. That will be easy to understand. @tfsumon

**Comment by devopstoday11** [comment]
@tfsumon 

one more idea to discuss, 

In the separate repos if I do this -
In compage-docs repo 
compage/version-number
changelog/version-number

can we get those into the main docs site repo as submodules? And in the live docs site, the user interactions should still be same, no change in UI interaction. 
@tfsumon

**Comment by tfsumon** [comment]
> > But it is possible if you organize all topic changelogs inside the 'content' folder instead of the 'changelog' folder.
> 
> could you please show me what you meant exactly? I didn't get it. give me example folder structure in this message how to organize the folders. That will be easy to understand. @tfsumon

Current folder structure:

1. `content/compage`
2. `content/changelog/compage`

If you reorganize the topics into the following structure, you can consolidate changelog and docs for 'compage' into the same repository:

1. `content/compage`
2. `content/compage-changelog`

@devopstoday11

**Comment by tfsumon** [comment]
> > But it is possible if you organize all topic changelogs inside the 'content' folder instead of the 'changelog' folder.
> 
> could you please show me what you meant exactly? I didn't get it. give me example folder structure in this message how to organize the folders. That will be easy to understand. @tfsumon

Current folder structure:

1. `content/compage`
2. `content/changelog/compage`

If you reorganize the topics into the following structure, you can consolidate changelog and docs for topic into the same repository:

1. `content/compage`
2. `content/compage-changelog`

@devopstoday11

**Comment by devopstoday11** [comment]
> If you reorganize the topics into the following structure, you can consolidate changelog and docs for topic into the same repository:
> 
> 1. `content/compage`
> 2. `content/compage-changelog`

So the new structure in the main docs site repo will be like this ?

`content/compage/version-number`
`content/compage-changelog/version-number`

@tfsumon

**Comment by tfsumon** [comment]
exactly

**Comment by devopstoday11** [comment]
@tfsumon with that new structure, there will not be any impact on the UI interaction right? still the flow will be same, correct?

**Comment by tfsumon** [comment]
Correct. But, when you click the 'changelog' button in the header, it will take you to a specific topic's changelog page rather than the root page for all topics.

The consolidated repo should have the following folders in the root:

1. `compage`
2. `compage-changelog`

**Comment by devopstoday11** [comment]
> Correct. But, when you click the 'changelog' button in the header, it will take you to a specific topic's changelog page rather than the root page for all topics.
> 
> separately consolidated repo should have the following folders in the root:
> 
> 1. `compage`
> 2. `compage-changelog`
> 
> @devopstoday11

Please proceed with this implementation with this new structure. 

we already have topics' structure in place.  you just have to structure the changelog folders and then show me a sample on how to use the separately consolidated repo , correct? @tfsumon

**Comment by tfsumon** [comment]
Before I proceed, could you set up the repository that will be added as a submodule in the main repository? I'm creating a sample repository for 'compage' using my GitHub account. Similarly, please create separate repositories for all other topics, including compage, using your GitHub account.

sample repo for compage - https://github.com/tfsumon/compage

**Comment by devopstoday11** [comment]
> Before I proceed, could you set up the repository that will be added as a submodule in the main repository? I'm creating a sample repository for 'compage' using my GitHub account. Similarly, please create separate repositories for all other topics, including compage, using your GitHub account.
> 
> sample repo for compage - https://github.com/tfsumon/compage
> 
> @devopstoday11
> 
> When you've completed it, please lmk.

I already created separate repos right - 

for example, https://github.com/intelops/compage-docs/ 
for all our projects I did create the repos like this - https://github.com/intelops/<projectname>-docs/

**Comment by tfsumon** [comment]
In your created repositories, there are currently no folders. Kindly consider adding following folders to each of your separate repositories.

<topic>
<compage>-changelog

@devopstoday11

**Comment by devopstoday11** [comment]
@tfsumon , 
please try with this sample for now and let me know if it's working or not -

 https://github.com/intelops/compage-docs/tree/chandu

**Comment by tfsumon** [comment]
Hello @devopstoday11!

Unfortunately, the current website structure with submodules isn't working. We need to make some changes:

**Possible Scenario:**
1. `compage-docs` submodule will be added to the main repo's `content` folder as `compage`.
2. `compage-changelog` submodule will also be added to the main repo's `content` folder as `compage-changelog`.

**Not Possible Scenario based on earlier discussed structure:**
1. `compage-docs` submodule containing both `docs` and `changelog` folders. Directly adding this submodule's content inside the `content` folder isn't possible without creating a folder inside `content` with a custom name or submodule name.

In this case, your main repo structure will look like:

- `content/<submodule or custom name>/compage`
- `content/<submodule or custom name>/compage-changelog`

If you want to keep both docs and changelog inside a single submodule, the main repo's structure will look like the second scenario I mentioned earlier. However, this structure might be challenging to work with given our current website flow. Let me know your preference. If separating them into two submodules works for you, we can follow the first scenario.

**Comment by devopstoday11** [comment]
@tfsumon ,
How about we structure like this?

the separate submodule related repo, for example - 
compage-docs repo will have folders structure like this 
` compage/<version-number>/<document-folders> `
in the same folder changelog will also be there like this 
` compage/<version-number>/changelog/ 

In the docs site main repo -
` content/compage ` - here compage folder is submodule 

So now in the docs site UI when user visits the site, the user will be able to see the changelog content under the documentation section and also separately under the changelog section as well. For both sections data display would be coming from same changelog folder. I assume here you probably need to develop the new flow for the docs site, correct? @tfsumon

**Comment by tfsumon** [comment]
> I assume here you probably need to develop the new flow for the docs site, correct?

Yeah. It won't take long since I've organized the code to easily modify the website flow whenever needed. So, your Git submodule repository structure will look like this: 

![image](https://github.com/intelops/external-docs/assets/114845583/fb110dbf-b76e-4eee-a32f-48f52ea5b0b5)

Please tell me if I should go ahead with this.

@devopstoday11

**Comment by devopstoday11** [comment]
> > I assume here you probably need to develop the new flow for the docs site, correct?
> 
> Yeah. It won't take long since I've organized the code to easily modify the website flow whenever needed. So, your Git submodule repository structure will look like this:
> 
> ![image](https://private-user-images.githubusercontent.com/114845583/285561451-fb110dbf-b76e-4eee-a32f-48f52ea5b0b5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExMzA4MTksIm5iZiI6MTcwMTEzMDUxOSwicGF0aCI6Ii8xMTQ4NDU1ODMvMjg1NTYxNDUxLWZiMTEwZGJmLWI3NmUtNGVlZS1hMzJmLTQ4ZjUyZWE1YjBiNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMTI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTEyOFQwMDE1MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNzFkZjQwYmY1Y2JhMDNlYzQxZWViMWY5ZTY1NzMyNzhhN2Y4NTdiYTlkM2I3Mjk0OGI4ZDY1YzZhMDRhZGVlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.r0p_vVvT6vQZGh2sRi79_V3I6AJ8P6nWgz0rlzLBjyA)
> 
> Please tell me if I should go ahead with this.
> 
> @devopstoday11

yes please proceed @tfsumon

**Comment by devopstoday11** [comment]
> Hi @devopstoday11,
> 
> Could you kindly reorganize the [repository](https://github.com/intelops/compage-docs/tree/chandu) folders according to the latest structure? Meanwhile, I'm actively working on developing the new flow for the documentation site.

@tfsumon , 
updated the repo -  https://github.com/intelops/compage-docs/tree/chandu

**Comment by devopstoday11** [comment]
@tfsumon , any updates?

**Comment by tfsumon** [comment]
Working on new flow—it's a bit tougher than expected. Hoping to wrap it up by tomorrow.

**Comment by devopstoday11** [comment]
> Hi @devopstoday11,
> 
> 
> 
> I apologize for the delay in completing the task. I encountered unexpected challenges, and I need one more day to ensure the quality of the work. I appreciate your understanding and patience.
> 
> 
> 
> Can you organize your Git submodule like this: https://github.com/tfsumon/compage-docs/tree/chandu? You can add a .md file, but it's recommended to avoid any extra files or folders.
> 
> 
> 
> For the changelog, visiting "siteurl.com/changelog" will display all changelogs (Like following image). 
> 
> 
> 
> ![image](https://github.com/intelops/external-docs/assets/114845583/086df8a9-ca04-4d6f-bfec-ca4c71e6e5b9)
> 
> 
> 
> Clicking a changelog link will lead to a detailed page. (Like following image)
> 
> 
> 
> ![image](https://github.com/intelops/external-docs/assets/114845583/071f5172-369a-4ae0-b2c8-d73a7c664613)
> 
> 
> 
> is it correct?

I'm confused. Please explain little bit more @tfsumon

**Comment by devopstoday11** [comment]
@tfsumon , pushed the restructured folders to the repo - https://github.com/intelops/compage-docs/tree/chandu

**Comment by tfsumon** [comment]
Thanks! I'll add this repo as git sub module and create a pull request once everything is finalized on my end.

**Comment by devopstoday11** [comment]
@tfsumon , done with this and also the URL Issue I mentioned on discord?
